addons/mail/models/mail_followers.py

```diff
@@ -124,16 +124,20 @@ def _get_recipient_data(self, records, message_type, subtype_id, pids=None, cids
         self.env['res.groups'].flush(['users'])
         self.env['mail.channel'].flush(['email_send', 'channel_type'])
         if records and subtype_id:
+            if pids:
+                pids_query = "UNION ALL SELECT id as partner_id, NULL as channel_id, FALSE as internal FROM (SELECT UNNEST(%s)) as tmp_partners(id)"
+            if cids:
+                cids_query = "UNION ALL SELECT NULL as partner_id, id as channel_id, FALSE as internal FROM (SELECT UNNEST(%s)) as tmp_channels(id)"
```

Comment:
If we use `UNION` rather than `UNION ALL` we do have a result that is more similar to what we had before, even despite the SELECT DISTINCT that surrounds everything. For example I think the `UNION ALL` will cause the `groups` array to contain duplicates if one of the `pids` happens to be a regular follower as well.

Reply:
the rationale behind "union all" VS "union" is to avoid having Postgresql de-duplicate records twice, when this de-duplication can be costly.

Reply:
But as we now make a `JOIN` instead of a `WHERE EXISTS`, we will have duplicated records in the query.

Ok, in this case, it won't be a problem due to the `DISTINCT` made in the upper query.